src/main/java/frc/robot/subsystems/elevator/Elevator.java

```diff
@@ -1,8 +1,108 @@
 package frc.robot.subsystems.elevator;
+
+import java.util.Optional;
+import java.util.TreeMap;
+
+import com.ctre.phoenix6.configs.Slot0Configs;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.configs.TalonFXConfigurator;
+import com.ctre.phoenix6.controls.Follower;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.util.sendable.SendableBuilder;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.drive.RobotDriveBase.MotorType;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
-import frc.robot.Constants;
 import frc.robot.RobotMap;
 
 public class Elevator extends SubsystemBase {
-    
+    public enum ElevatorHeight {
+        HOME,
+        SCORE_L4,
+        SCORE_L3,
+        SCORE_L2,
+        SCORE_L1,
+        PRE_SHALLOWCLIMB,
+        SCORE_SHALLOWCLIMB;
+    }
+
+   
+    private TreeMap<ElevatorHeight, Double> elevatorPositions = new TreeMap<ElevatorHeight, Double>();
+                
+    private ElevatorHeight desiredPosition = ElevatorHeight.HOME;
+
+    public TalonFX leftMotor = new TalonFX(RobotMap.LEFT_ELEVATOR_CAN_ID);
+    public TalonFX rightMotor = new TalonFX(RobotMap.RIGHT_ELEVATOR_CAN_ID);
+   
+    final TrapezoidProfile m_profile = new TrapezoidProfile(
+        new TrapezoidProfile.Constraints(ElevatorCal.MAX_VELOCITY_IN_PER_SECOND, ElevatorCal.MAX_ACCELERATION_IN_PER_SECOND_SQUARED)
+    );
+    private TrapezoidProfile.State m_goal = new TrapezoidProfile.State();
+    private TrapezoidProfile.State m_setpoint = new TrapezoidProfile.State();
+
+    public Elevator() {
+
+        elevatorPositions.put(ElevatorHeight.HOME, ElevatorCal.POSITION_HOME_INCHES);
+        elevatorPositions.put(ElevatorHeight.SCORE_L4, ElevatorCal.POSITION_SCORE_L4_INCHES);
+        elevatorPositions.put(ElevatorHeight.SCORE_L3, ElevatorCal.POSITION_SCORE_L3_INCHES);
+        elevatorPositions.put(ElevatorHeight.SCORE_L2, ElevatorCal.POSITION_SCORE_L2_INCHES);
+        elevatorPositions.put(ElevatorHeight.SCORE_L1, ElevatorCal.POSITION_SCORE_L1_INCHES);
+        elevatorPositions.put(ElevatorHeight.PRE_SHALLOWCLIMB, ElevatorCal.POSITION_PRE_SHALLOWCLIMB_INCHES);
+        elevatorPositions.put(ElevatorHeight.SCORE_SHALLOWCLIMB, ElevatorCal.POSITION_SCORE_SHALLOWCLIMB_INCHES);
+        initTalons();
+    }
+
+    private void initTalons() {
+        var slot0Configs = new Slot0Configs();
+        TalonFXConfigurator cfgLeft = leftMotor.getConfigurator();
+        TalonFXConfigurator cfgRight = rightMotor.getConfigurator();
```

Comment:
this isn't needed because you later set the right motor to follow the left, so it will automatically match configurations